src/Cache/Engine/RedisClusterEngine.php

```diff
@@ -0,0 +1,195 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.2.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\Cache\Engine;
+
+use Cake\Core\Exception\CakeException;
+use Cake\Log\Log;
+use Redis;
+use RedisCluster;
+use RedisClusterException;
+
+/**
+ * Redis cluster cache storage engine.
+ */
+class RedisClusterEngine extends RedisEngine
```

Comment:
Might make the code bit messy but we could create RedisEngine as cluster instance if nodes array is given.
Then we would need to check if instance is RedisCluster in few places.

Reply:
Instead of inheriting, we could clone + modify `RedisEngine` into `RedisClusterEngine`. I think the differences in method support warrant a separate implementation that can evolve separately as redis cluster changes.

Reply:
@markstory Looking at the updated `RedisEngine` which supports redis cluster too, having two separate engine classes would lead to a lot of code duplication. Apart from the connection code, only 2 other methods needed changes. So if we want to keep separate classes then we should use a trait for the common parts.

Reply:
Ok. Lets leave this as it is then. :+1: